controllers/cloud.redhat.com/namespacereservation_controller.go

```diff
@@ -187,6 +190,20 @@ func (r *NamespaceReservationReconciler) Reconcile(ctx context.Context, req ctrl
 			return ctrl.Result{}, err
 		}
 
+		duration, err := ParseDurationTime(res.Spec.Duration)
+		if err != nil {
+			r.Log.Error(err, "Cannot parse duration")
+		}
+
+		averageRequestedDurationMetrics.With(prometheus.Labels{"controller": "namespacereservation"}).Observe(float64(duration.Hours()))
+
+		defer func() {
+			t := time.Now()
+
+			elapsed := t.Sub(start)
```

Comment:
I'm not 100% sure of the purpose of this metric?

Reply:
I removed that defer. I have it so that the timer starts at the beginning of the reconcile function and stops at the end.